### PR TITLE
Feature/badges url

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java [hunk 1]
```diff
@@ -75,7 +83,7 @@ public DescriptorImpl getDescriptor() {
   @Override
   public OutputStream decorateLogger(AbstractBuild build, OutputStream outputStream) throws IOException, InterruptedException {
     SonarInstallation inst = SonarInstallation.get(getInstallationName());
-    if (inst == null || inst.isDisabled() || (inst.getDatabasePassword() == null && inst.getSonarPassword() == null)) {
```
Comment (henryju) [inline src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java]: Is this really related to the same ticket? ;)

Comment (dbmeneses) [inline src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java]: There are 2 commits in this P/R, but it is the right ticket. Sorry for the mess

**Comment by henryju** [comment]
LGTM appart build failure

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 2]
```diff
@@ -284,6 +285,13 @@ private void handleErrors(Run<?, ?> build, TaskListener listener, SonarRunnerIns
       errorMessage += Messages.SonarRunner_GlobalConfigNeeded();
     }
     e.printStackTrace(listener.fatalError(errorMessage));
+
+    try {
+      SonarUtils.addUrlActionTo(build);
+    } catch (IOException e1) {
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: CRITICAL Either log or rethrow this exception. [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1166)

File: src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java [hunk 3]
```diff
@@ -284,6 +285,13 @@ private void handleErrors(Run<?, ?> build, TaskListener listener, SonarRunnerIns
...
+      e1.printStackTrace(listener.fatalError(errorMessage));
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarRunnerBuilder.java]: CRITICAL Use a logger to log this exception. [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java [hunk 4]
```diff
@@ -168,6 +183,18 @@ public void buildEnvVars(Map<String, String> env) {
...
       }
     }
+    
+    @Override
+    public boolean tearDown(AbstractBuild build, BuildListener listener) throws IOException, InterruptedException {
+      UrlSonarAction urlAction = SonarUtils.addUrlActionTo(build);
+      
+      // null result means success so far
+      if (urlAction != null && urlAction.isNew() && build.getResult() == null  && build.getAction(BuildSonarAction.class) == null) {
+          build.addAction(new BuildSonarAction(urlAction.getSonarUrl()));
```
Comment (SonarTech) [inline src/main/java/hudson/plugins/sonar/SonarBuildWrapper.java]: MAJOR Make this line start at column 9. [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 6 issues:
* BLOCKER 2 blocker
* CRITICAL 2 critical
* MAJOR 2 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [NullPointerException might be thrown as 'getInstance' is nullable here](https://github.com/SonarSource/jenkins-sonar-plugin/blob/985279e72e504bf3f12138710f4e9494e483eba4/src/main/java/hudson/plugins/sonar/action/BuildSonarAction.java#L74) [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS2259)
* BLOCKER [NullPointerException might be thrown as 'getInstance' is nullable here](https://github.com/SonarSource/jenkins-sonar-plugin/blob/985279e72e504bf3f12138710f4e9494e483eba4/src/main/java/hudson/plugins/sonar/action/ProjectSonarAction.java#L60) [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS2259)
* MAJOR Add a 'package-info.java' file to document the 'sonar' package (org.jenkins-ci.plugins:sonar:target/generated-sources/localizer/hudson/plugins/sonar) [rule](https://dory.internal.sonarsource.com/coding_rules#rule_key=squid%3AS1228)
